test/monitoring/test.py

```diff
@@ -0,0 +1,128 @@
+"""Smoke test for monitoring API."""
+
+from google.cloud.gapic.monitoring.v3 import group_service_api
+from google.cloud.gapic.monitoring.v3 import metric_service_api
+from google.cloud.gapic.monitoring.v3 import enums
+from google.gax import CallOptions, INITIAL_PAGE
+from google.monitoring.v3 import group_pb2
+from google.monitoring.v3 import common_pb2
+from google.monitoring.v3 import metric_pb2
+from google.protobuf import timestamp_pb2
+from google.api import metric_pb2 as api_metric_pb2
+
+PROJECT_ID = '<PROJECT_ID>'
+
+def testGroupCollection():
+  api = group_service_api.GroupServiceApi()
+  projectName = api.project_path(PROJECT_ID)
+  group = group_pb2.Group(
+    display_name='displayName', filter='resource.type = gce_instance')
+  createdGroup = api.create_group(projectName, group)
+
+  print "New group created: \n%s\n" % createdGroup
+
+  groupName = createdGroup.name
+  groupToUpdate = group_pb2.Group(name=groupName, display_name='displayName2',
+    filter='resource.type = gce_instance')
+  updatedGroup = api.update_group(groupToUpdate)
+  print "Group updated: \n%s\n" % updatedGroup
+
+  # group id is the last segment in the returned group name.
+  groupId = updatedGroup.name[createdGroup.name.rfind('/')+1:]
+  api = group_service_api.GroupServiceApi()
+  name = api.group_path(PROJECT_ID, groupId)
+  response = api.get_group(name)
+  print "Group fetched: \n%s\n" % response
+
+  print "List groups:\n"
+  for group in api.list_groups(name):
+      # process element
+      print(group)
+
+  print "List group members:\n"
+  name = api.group_path(PROJECT_ID, groupId)
+  for member in api.list_group_members(
+    name=name,
+    interval=common_pb2.TimeInterval(
+      end_time=timestamp_pb2.Timestamp(seconds=1475262206))):
+    print(member)
+
+  print "Delete groups:\n"
+  name = api.group_path(PROJECT_ID, groupId)
+  api.delete_group(name)
+  try:
+    response = api.get_group(name)
+  except:
+    print "Expect error here since the group should have been deleted"
+
+def testMetricCollection():
+  custom_metric_type = 'custom.googleapis.com/custom/instance/cpu/utilization'
+  api = metric_service_api.MetricServiceApi()
+  name = api.project_path(PROJECT_ID)
+
+  metric_descriptor = api_metric_pb2.MetricDescriptor(
+    type=custom_metric_type,
+    metric_kind=enums.MetricDescriptor.MetricKind.GAUGE,
+    value_type=enums.MetricDescriptor.ValueType.BOOL)
+  response = api.create_metric_descriptor(name, metric_descriptor)
+  print "Metric created: \n%s\n" % response
+
+  # This part is flaky as the backend will sometimes return NOT_FOUND for the
+  # metric created in last step.
```

Comment:
Put a sleep in here, like @geigerj did in https://github.com/googleapis/api-client-staging/pull/78